pkg/gvk/gvk.go

```diff
@@ -300,14 +485,19 @@ func (r *Compare) compareResources(src []*metav1.APIResourceList, dst []*metav1.
 	SortResources(src)
 	for _, srcList := range src {
 		missing := []metav1.APIResource{}
+		matchingDstResourceList := findResourceList(srcList.GroupVersion, dst)
 		for _, resource := range srcList.APIResources {
 			if cohabitator, found := r.CohabitatingResources[resource.Name]; found {
-				if cohabitator.Seen {
-					continue
+				gv, err := schema.ParseGroupVersion(srcList.GroupVersion)
+				if err != nil &&
+					(gv.Group == cohabitator.groupResource1.Group || gv.Group != cohabitator.groupResource2.Group) {
```

Comment:
little confused why this is required. @sseago can you elaborate on what is happening here?

Reply:
@alaypatel07 The previous code identified cohabitating resources by the two groups they cohabitated in, but the code here that identified cohabitators and passed on them didn't bother to check the group. If I were to create a CRD that I call "deployments" in my own "sseago.example.com" group, it would be flagged as cohabitating with "deployments.extensions" and "deployments.apps", which seems like a bug. This change checks the two groupresource fields (`groupResource1` and `groupResource2` ) to make sure that we only flag cohabitating resources that actually belong to the cohabitating groups. There was actually no code in mig-controller that read the values of these two fields until I added this check.

Reply:
I need to check the unit tests -- looks like this change may have broken a test.

Reply:
ack, I think you might want

```
				if err == nil &&
					(gv.Group == cohabitator.groupResource1.Group || gv.Group == cohabitator.groupResource2.Group) 
```

Reply:
also, @sseago if you are fixing this bug here, likely means we need the same bug fix in velero as well.

https://github.com/vmware-tanzu/velero/blob/bd5ba66e42e48ca0338e01d7c2d4375eea68e052/pkg/backup/item_collector.go#L211

Reply:
@alaypatel07 Good catch on the backwards err nil check. Yes, if we succeed in parsing GV, then we use GV to compare with cohabitator Group. And yeah, I'll create an issue and PR upstream in velero. I'd forgotten that this code was taken from Velero. They're also not using the cohabitator groupResources for anything (other than logging).

Reply:
Ugh. I shouild have said good catch on *both* backwards checks. Turns out the failing unit test actually did the right thing. Fixing both of these problems made the existing unit test work succeed again.

Reply:
+1, I should have pointed out both the errors. I was a little confused as to how best to communicate that so ended up pasting the exact string that worked